test/parallel_api/ranges/std_ranges_test.h

```diff
@@ -144,22 +143,34 @@ template<typename T>
 static constexpr
 bool is_range<T, std::void_t<decltype(std::declval<T&>().begin())>> = true;
 
+using AllHostPolicies = std::integral_constant<int, 0>;
+using ParHostPolicies = std::integral_constant<int, 1>;
+using SkipHostPolicies = std::integral_constant<int, 2>;
+
 template<typename DataType, typename Container, TestDataMode test_mode = data_in>
 struct test
 {
     const int max_n = 10;
-    template<typename Policy>
-    std::enable_if_t<std::is_same_v<Policy, std::true_type>>
-    operator()(Policy, auto algo, auto& checker, auto... args)
+    void operator()(AllHostPolicies, auto algo, auto& checker, auto... args)
     {
         operator()(oneapi::dpl::execution::seq, algo, checker, args...);
         operator()(oneapi::dpl::execution::unseq, algo, checker, args...);
         operator()(oneapi::dpl::execution::par, algo, checker,  args...);
         operator()(oneapi::dpl::execution::par_unseq, algo, checker, args...);
     }
+    void
+    operator()(ParHostPolicies, auto algo, auto& checker, auto... args)
+    {
+        operator()(oneapi::dpl::execution::par, algo, checker,  args...);
+        operator()(oneapi::dpl::execution::par_unseq, algo, checker, args...);
+    }
+    void
+    operator()(SkipHostPolicies, auto algo, auto& checker, auto... args) {}
 
     template<typename Policy, typename Algo, typename Checker, typename TransIn, typename TransOut, TestDataMode mode = test_mode>
-    std::enable_if_t<!std::is_same_v<Policy, std::true_type> && mode == data_in>
+    std::enable_if_t<mode == data_in && !std::is_same_v<Policy, AllHostPolicies> &&
```

Comment:
The condition
```C++
!std::is_same_v<Policy, AllHostPolicies> &&
                                               !std::is_same_v<Policy, ParHostPolicies> &&
                                               !std::is_same_v<Policy, SkipHostPolicies>
```
has used 5 times in your changes.

What about to pack it into some useful type?

Reply:
Thanks, I've added a trait